src/traversing.js

```diff
@@ -143,7 +143,15 @@ jQuery.each( {
 		return siblings( elem.firstChild );
 	},
 	contents: function( elem ) {
-		return elem.contentDocument || jQuery.merge( [], elem.childNodes );
+        if ( jQuery.nodeName( elem, "iframe" ) ) {
+            return elem.contentDocument;
+        }
+
+        if ( jQuery.nodeName( elem, "template" ) ) {
+            elem = elem.content || elem;
```

Comment:
@TechQuery All places where a specific logic has to be applied because of deficiencies of a browser a support comment needs to be added like [this one](https://github.com/jquery/jquery/blob/2d4f53416e5f74fa98e0c1d66b6f3c285a12f0ce/src/data.js#L110).

In this case, I'd expect something like that:
```js
// Support: IE 9 - 11 only, iOS 7 only, Android Browser <=4.3 only
// Treat the template element as a regular one in browsers that
// don't support it.
```
It's only necessary to mention browsers we still support; see https://jquery.com/browser-support/.